src/PollingBlockTracker.test.ts

```diff
@@ -185,6 +185,33 @@ describe('PollingBlockTracker', () => {
       );
     });
 
+    it('should not retry failed requests after the block tracker is stopped', async () => {
```

Comment:
Hmm, this test does not test that requests are not retried. What it tests is that the promise that `getLatestBlock` returns eventually resolves or is rejected. Should we be more clear here?

``` suggestion
    it('should return a promise that rejects if not resolved by the time the block tracker is stopped', async () => {
```

Or do we have a way to test that requests are not retried?

Reply:
Well, perhaps the original test name is fine. I guess the test would fail if the polling loop continued because we are using `numAutomaticCalls: 1`.

Reply:
Hmm, I think you are right, this test doesn't check for how many provider requests are fired, so even if there is a similar outcome the case name may be wrong. I'm inclined to commit your suggestion and add another test case specific for spying on provider requests

Reply:
Added in [efceaf7](https://github.com/MetaMask/eth-block-tracker/pull/284/commits/efceaf7e3ca34fcc4d4eb0268a0fd339dc423c01)